src/libNode/MicroBlockPreProcessing.cpp

```diff
@@ -679,10 +679,16 @@ bool Node::RunConsensusOnMicroBlockWhenShardLeader() {
 
   if (m_mediator.m_ds->m_mode == DirectoryService::Mode::IDLE &&
       !m_mediator.GetIsVacuousEpoch()) {
+    unsigned int sleep_time =
+        TX_DISTRIBUTE_TIME_IN_MS +
+        ((m_mediator.m_dsBlockChain.GetLastBlock().GetHeader().GetEpochNum() ==
```

Comment:
Is this line to check if we are in the first Tx epoch after a new DS epoch?  Doesn't seem correct.